typewriter/fixes/fixer_utils.py

```diff
@@ -133,3 +138,180 @@ def touch_import(package, name, node):
+            return find_import_binding_info(name, child, package, return_if_aliased)
+    return None
+
+
+def find_import_binding(name, node, package=None, return_if_aliased=False):
+    # type: (str, Node, str, bool) -> Optional[Node]
+    """
+    Returns just the node from the find_import_binding_info call
+
+    Parameters
+    -----------
+    name : str
+    node : Node
+    package : str
+    return_if_aliased : bool
+
+    Return
+    -----------
+    Optional[Node]
+    """
+    result = find_import_binding_info(name, node, package, return_if_aliased)
+    if result:
+        return result.node
+    return None
+
+
+def _is_dotted_import(node, name, package=None):
+    """
+    Returns node if the node represents an import of the name (or package.name). This helper
+    can be used for both dotted and non-dotted names, i.e. import a.b or import a respectively
+    """
+    if str(node).strip() == name:
+        return node
+
+    if node.type == syms.dotted_name:
+        if package and not str(node.children[:-1]).strip() == package:
+            return None
+        if str(node.children[-1]).strip() == name:
+            return node
+
+    return None
+
+
+def _get_import_binding_info(node, name, package=None, return_if_aliased=False):
+    # type: (Node, str, str, bool) -> Optional[ImportInfo]
+    """
+    Will return node if node will import name, or node
+    will import * from package.  If return_if_aliased is
+    True then returns binding even if the binding is for an alias of name
+    in the import:
+        i.e. will return "import <name> as <alias>" or "from <package> import <name> as <alias>
+
+    None is returned otherwise.
+
+    Will always return None for a statement of the form
+        "from X import Y as Z"
+    if the <package> argument is passed in. See lib2to3.tests.test_util for explanation
+    See test cases for examples.
+
+    Parameters
+    -----------
+    node : Node
+    name : str
+    package : str
+    return_if_aliased : bool
+
+    Return
+    -----------
+    Optional[ImportInfo]
+    """
+    def handle_as_name(name, node):
+        if node.type in (syms.import_as_name, syms.dotted_as_name):
+            imp = node.children[0]
+            alias = node.children[-1]
+            if (alias.type == token.NAME and alias.value == name) or \
+                    (_is_dotted_import(imp, name) and return_if_aliased):
+                # Sometimes there is a leading whitespace, so to be safe we strip
+                return ImportInfo(str(alias).strip(), str(imp).strip(), node)
+        elif node.type == token.NAME and node.value == name:
+            return ImportInfo(name, name, node)
+
+    if node.type == syms.import_name and not package:
+        imp = node.children[1]
+        if imp.type == syms.dotted_as_names:
+            for child in imp.children:
+                result = handle_as_name(name, child)
+                if result:
+                    return result
```

Comment:
this is because `imp`  might be `dotted_as_names` as in `import x.y as z, a.b as c` or `dotted_as_name` as in `import x.y as z`, the `handle_as_name` below covers the latter and the former is just the same thing but iterating over each of the imports individually

Reply:
ah I see, I misunderstood. Yeah I agree that makes sense